libcxx/test/support/invocable_with_telemetry.h

```diff
@@ -0,0 +1,100 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TEST_SUPPORT_INVOCABLE_WITH_TELEMETRY_H
+#define TEST_SUPPORT_INVOCABLE_WITH_TELEMETRY_H
+
+#include <cassert>
+#include <concepts>
+#include <utility>
+
+#if TEST_STD_VER < 20
+#  error invocable_with_telemetry requires C++20
+#else
+template <class F>
+class invocable_with_telemetry {
```

Comment:
I have a bunch of questions that stem from why/if we really need to pass around three separate addresses, one per metric, or if we can simply pass around a pointer to struct containing whatever we want?

Reply:
I've added `invocable_telemetry` to structure this a bit better.